Interfaces first migration

ocket8888 opened the pull request:
## What does this PR (Pull Request) do?
- [x] This PR fixes #4611 


## Which Traffic Control components are affected by this PR?
- Traffic Ops (database)

## What is the best way to verify this PR?
Run a fresh installation of Traffic Ops and the database, verify that the new tables exist.

Run the migration against a database with existing data, verify that tables are properly populated and nothing old is mutated.

## The following criteria are ALL met by this PR

- [x] We don't have database tests
- [x] Documentation is unnecessary
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

shamrickus commented on traffic_ops/app/db/migrations/20200414000000_interfaces.sql:
```diff
@@ -0,0 +1,151 @@
+
+-- +goose Up
+CREATE TABLE interface (
+	max_bandwidth bigint DEFAULT NULL CHECK (max_bandwidth IS NULL OR max_bandwidth >= 0),
```
Blueprint specifies that the default for max_bandwidth should be 0

ocket8888 replied:
Yeah, the intention was for the default to be "no limit", but since writing the blueprint the sentiment has been that instead `0` should literally just mean `0` and `null` should mean "no limit".

ocket8888 commented:
This isn't ready to merge yet, my company has some production data that includes subnets in gateways, which under this model is not allowed. So I'm waiting to hear back about their usage and necessity.

ocket8888 commented:
Okay, it was determined those are probably mistakes, so I added some logic to strip them on upgrades. Plus I fixed a check constraint to properly consider mask length based on address family.

mattjackson220 commented on traffic_ops/app/db/migrations/20200414000000_interfaces.sql:
```diff
@@ -0,0 +1,159 @@
+				) AS digits)
+		END
+	),
+	NULLIF(regexp_replace(server.ip_gateway, '/\d+$', '')::inet,
```
this needs `, '')` before the ::inet

mattjackson220 commented:
Looks good! creates / inserts successfully, reverts successfully, old data is unchanged
